TypeCobol.LanguageServer/TypeCobolCustomLanguageServer/TypeCobolCustomLanguageServer.cs

```diff
@@ -23,29 +24,46 @@ public TypeCobolCustomLanguageServer(IRPCServer rpcServer, Queue<MessageActionWr
         /// </summary>
         public bool UseOutlineRefresh { get; set; }
 
-        protected override void OnShutdown()
-        {
-            if (UseOutlineRefresh && this.Workspace.DocumentModifiedEvent != null)
-                this.Workspace.DocumentModifiedEvent -= DocumentModified;
+        /// <summary>
+        /// Are we using the CFG view in the client.    
+        /// </summary>
+        public bool UseCfgDfaDataRefresh { get; set; }
+        /// <summary>
+        /// True if The server is executed in LSR TestMode
+        /// </summary>
+        public bool InRobotLsrTestMode { get; set; }
+        /// <summary>
+        /// True if the server is executed in LSR client mode.
+        /// </summary>
+        public bool InLsrClientMode { get; set; }
 
-            base.OnShutdown();
+        protected override InitializeResult OnInitialize(InitializeParams parameters)
+        {
+            var result = base.OnInitialize(parameters);
+            this.Workspace.DocumentModifiedEvent += DocumentModified;
+            this.Workspace.ClientConfigurationChangedEvent += OnClientOptionsChanged;
+            return result;
         }
 
-        protected override void OnDidOpenTextDocument(DidOpenTextDocumentParams parameters)
-        {
-            if (UseOutlineRefresh && this.Workspace.DocumentModifiedEvent == null)
+        /// <summary>
+        /// Handle -ol and -cfg options from the client configuration change notification.
+        /// </summary>
+        /// <param name="sender">Should be the Workspace instance</param>
+        /// <param name="options">Client's Options</param>
+        private void OnClientOptionsChanged(object sender, IEnumerable<string> options)
+        {            
+            if (!this.InRobotLsrTestMode)
```

Comment:
Can you explain why in Robot mode we should not update the options ?

Reply:
Because each LSR test are configured differently, and -ol and -cfg options come as command line options not as configuration changed options. LSR tests create a custom Initialize and Configuration Change notification. Maybe we should review this aspect.

Reply:
But you are right for -ol and -cfg options inside LSRTest, I am going to parameterized these options in the custom config.json generated.